holoviews/core/spaces.py

```diff
@@ -413,8 +416,22 @@ class Callable(param.Parameterized):
     inputs = param.List(default=[], doc="""
          The list of inputs the callable function is wrapping.""")
 
+    def __init__(self, **params):
+        super(Callable, self).__init__(**params)
+        self._memoized = {}
+
     def __call__(self, *args, **kwargs):
-        return self.callable_function(*args, **kwargs)
+        inputs = [inp for inp in self.inputs if isinstance(inp, DynamicMap)]
```

Comment:
Very minor point - I would call the loop variable ``i`` given that ``inp`` is a bit weird and you probably don't want to clash with the ``input`` built-in. Alternatively, this is one instance of a pure ``filter`` so you could consider using that (it would be shorter).